tests/tests.rs

```diff
@@ -0,0 +1,254 @@
+#[cfg(test)]
+mod tests {
+	use mockito::{self, Matcher};
+	use prost::Message;
+	use std::future::Future;
+	use vss_client::client::VssClient;
+	use vss_client::error::VssError;
+
+	use vss_client::types::{
+		ErrorCode, ErrorResponse, GetObjectRequest, GetObjectResponse, KeyValue, ListKeyVersionsRequest,
+		ListKeyVersionsResponse, PutObjectRequest, PutObjectResponse,
+	};
+
+	const GET_OBJECT_ENDPOINT: &'static str = "/getObject";
+	const PUT_OBJECT_ENDPOINT: &'static str = "/putObjects";
+	const LIST_KEY_VERSIONS_ENDPOINT: &'static str = "/listKeyVersions";
+
+	#[tokio::test]
+	async fn test_get() {
+		// Spin-up mock server with mock response for given request.
+		let base_url = mockito::server_url().to_string();
+
+		// Set up the mock request/response.
+		let get_request = build_get_request();
+		let mut mock_response = GetObjectResponse::default();
+		mock_response.value = Some(KeyValue { key: "k1".to_string(), version: 2, value: b"k1v2".to_vec() });
+
+		// Register the mock endpoint with the mockito server.
+		let mock_server = mockito::mock("POST", GET_OBJECT_ENDPOINT)
+			.match_body(get_request.encode_to_vec())
+			.with_status(200)
+			.with_body(mock_response.encode_to_vec())
+			.create();
+
+		// Create a new VssClient with the mock server URL.
+		let vss_acc = VssClient::new(&base_url);
+
+		let actual_result = vss_acc.get_object(&get_request).await.unwrap();
+
+		let expected_result = &mock_response;
+		assert_eq!(&actual_result, expected_result);
```

Comment:
nit: little more idiomatic to dereference `expected_result` as otherwise they'll both need to be dereferenced anyhow.